### DISCUSSION TITLE
Advice needed - build "table rows" from repeater+container(text+checkbox) and forward "selected rows" to automation

### DISCUSSION THREAD

**Discussion by pandel** [Budibase/budibase#7453] [Help] [answered]
Hi!

As it is impossible to use row selection with a table build upon a query data source, I tried to rebuild the whole thing like this:

```
Form
-> Data Provider
--> Repeater
----> Container
--------> Checkbox
--------> Paragraph (Field1)
--------> Paragraph (Field2)
--------> ...
```

I then added the following actions to the checkbox:
1. Update State / Set value / Key: ROW_SELECT / Value:
```
if (Array.isArray($("State.ROW_SELECT"))) {
	let arr=$("State.ROW_SELECT");
} else {
	let arr = [];
}
if ($("Field Value")==true) {
  arr.push($("Repeater.Ausgeschlossene Meldungen.id"));
} else {
  var index = arr.indexOf($("Repeater.Ausgeschlossene Meldungen.id")); 
  if (index !== -1){
    arr.splice(index, 1);
  }
}
return arr;
```

The main button, which starts the automation sends `State.ROW_SELECT` to the automation.

The automation itself has a field, type array, that should receive `State.ROW_SELECT` and use looping to process any further. ATM I simply try to send `{{ loop.currentItem }}` via email to me, just to see if the automation receives anything.

But all that does not seem to work, I only get an email with "Error while executing JS" as text. And I don't know how and where to debug the JS used here.

Is it somehow possible to achieve what I am trying to do? I am definitely no JavaScript guy and so, I need a little more advice.

Thanks,
Holger

**Comment by pandel** [comment, marked as answer]
I have to answer myself :-) - my Javascript was wrong, as expected! It works like this as `Update State` checkbox action:

```
let a=$("State.ROW_SELECT");
if (Array.isArray(a)) {
	if ($("Field Value") == true) {
		a.push($("Repeater.Ausgeschlossene Meldungen.id"));
		return a;
	} else {
		var index = a.indexOf($("Repeater.Ausgeschlossene Meldungen.id")); 
		if (index !== -1){
			a.splice(index, 1);
			return a;
		}
	}
} else {
	if ($("Field Value") == true) {
		return [$("Repeater.Ausgeschlossene Meldungen.id")];
	} else {
		return [];
	}
}
```

Now I can take the table row ID inside the automation and do whatever I like in the background!

**Reply by kellis5137** [reply]
Hi,
I was looking to do something similar (multi-select from an existing table to send to an automation) and basically did exactly what it appears you did here. When I select one checkbox, it grabs the id. However, when I grab others and click submit, it doesn't fill up ROW_SELECT and it doesn't send that array to the automation; just the first selection. Also, the way the browser reacts, it appears the script runs for the first checkbox selected, but fails for any second, third, etc. I don't think it's your logic, it seems sound. However, I'm just wondering, did you have to do anything else outside of what you had listed here?

**Comment by pandel** [comment]
@kellis5137 Hey sorry, have been on holdiay... you have to make sure that you components are ordered correctly. See my comment here: https://github.com/Budibase/budibase/issues/7668#issuecomment-1239319353
